### ISSUE TITLE
Getting "File is not a module" error for file exporting SVG

### ISSUE THREAD

**Issue by TheKvist** [TypeStrong/ts-loader#1113]
I am experiencing quite a strange issue, I've been unable to resolve. I am trying to extract my React components from their main application into a library and want to bundle it with Webpack, as some of the components are using SVG assets. Also, I want to export these assets themselves, too, so client applications can use both, components and individual icons from the library.

For the SVGs, I have copied our `babel-loader` configuration, which uses `babel-plugin-named-asset-import` in order to load the SVGs using `@svgr/webpack`. Of course, I still need to generate the Typescript declarations to import the actual components in a client application and added a minimal `ts-loader` configuration right before the `babel-loader` configuration. My whole webpack.config.js looks like this:

```js
module.exports = (env, args) => {
  return {
    entry: "./index.ts",
    module: {
      rules: [
        {
          test: /\.tsx?$/,
          use: {
            loader: "ts-loader",
          },
          exclude: /node_modules/,
        },
        {
          test: /\.(js|mjs|jsx|ts|tsx)$/,
          loader: require.resolve("babel-loader"),
          options: {
            plugins: [
              [
                require.resolve("babel-plugin-named-asset-import"),
                {
                  loaderMap: {
                    svg: {
                      ReactComponent: "@svgr/webpack?-svgo,+ref![path]",
                    },
                  },
                },
              ],
            ],
          },
        },
      ],
    },
    resolve: {
      extensions: [".tsx", ".ts", ".js"],
    },
  };
};
```

My project structure looks like this: 

```sh
.
├── config
│   └── webpack.config.js
├── index.ts
├── package.json
├── package-lock.json
├── src
│   ├── assets
│   │   └── icons
│   │       ├── circle.svg
│   │       └── index.ts
│   ├── demo
│   │   └── index.ts
│   └── global.d.ts
└── tsconfig.json
```

Code-wise there really is nothing out of the ordinary here.

In `./src/assets/icons/index.ts`, we have this single line:
```ts
export { ReactComponent as Circle} from "./circle.svg";
```

SVG is declared as a module in `./global.d.ts` like this (copied from the source project, where everything works):

```ts
/// <reference types="react" />

declare module "*.svg" {
  import * as React from "react";

  export const ReactComponent: React.FunctionComponent<React.SVGProps<SVGSVGElement>>;

  const src: string;
  export default src;
}
```

Then, for the code, in `./src/demo/index.ts` we have this:
```ts
import { Circle} from "../assets/icons";

export default {
  Circle,
};
```

`./index.ts` contains the exact same, because I realized that compilation works for this file, but not the other one.

### Expected Behaviour

Running `npx webpack --config ./config/webpack.config.js` will use `ts-loader`, which in turn starts `tsc` to compile the Typescript and generate declarations, then the SVGs will be loaded by `babel-loader` so everything is correctly bundled.

### Actual Behaviour

The build immediately fails with this error:

```sh
ERROR in [root]/src/demo/index.ts
[tsl] ERROR in [root]/tstest/src/demo/index.ts(1,21)
      TS2306: File '[root]/tstest/src/assets/icons/index.ts' is not a module.
```

From my understanding, this error gets thrown if the file does not contain the "export" keyword and I wonder why, since "export" is literally the very first word in that file.

Two things about this are extremely puzzling to me:

1. Simply running `tsc` compiles all the files perfectly fine, but obviously without the SVGs. I wonder what `ts-loader` is doing different from standalone `tsc` here?
2. I have **two** files containing the exact same code (copy-pasted to make sure). `./index.ts` and `./src/demo/index.ts`, yet the error gets thrown for only `./src/demo/index.ts`. Unfortunately, this file basically represents real code as it appears in the components I am trying to extract.

In the actual library, I get the error for every single source file that is referencing the `./src/assets/icons/index.ts` file.

### Steps to Reproduce the Problem

1. Clone the demo repo
2. Run `npm install`
3. Run `npx webpack --config ./config/webpack.config.js`
(4. Run `tsc` for comparison and/or delete the `demo` directory and run webpack again)

### Location of a Minimal Repository that Demonstrates the Issue.

https://github.com/TheKvist/ts-loader-example

**Comment by appzuka** [comment]
Do you need to include all source files in tsconfig.json?

```
  "include": ["./index.ts", "src/**/*"],
```

if you change this to:

````
  "include": ["src/**/*.d.ts"],
````

then you don't get the error.  In this case, src/demo.ts will not be compiled, but if you import it from the main index.ts then it will, without error.  This is my understanding of how webpack works - it includes all source files referenced from the entry point.  There is no need to include all source files in the tsconfig.json as well.

If you need both files to produce separate output you can include 2 entry points:

```
    entry: {main: "./index.ts", second: "./src/demo/index.ts"},
```
I guess what is happening is that if you don't include demo/index.ts in tsconfig.json then it is compiled through webpack, which runs it through @svgr/webpack before typescript.  If you do include it in tsconfig then typescript tries to compile it before @svgr/webpack, which produces an error.

**Comment by TheKvist** [comment]
@appzuka Yes, of course! This way, it makes perfect sense. Besides from me not fully having had my head wrapped around webpack, I think I got additionally confused when Typescript was introduced into the whole deal. I've never really worked longer than a few weeks at a time with webpack before and never at all with Typescript, so I somehow lost track of what was happening. Thanks a lot for clearing that up.
I changed the tsconfig.json in the real library accordingly and it fully compiled for the first time. You really saved my day, good sir.
Closing the issue as the problem is resolved.
